packages/palette/src/elements/Modal/Modal.tsx

```diff
@@ -51,11 +55,34 @@ export const Modal: SFC<ModalProps> = ({
   })
   const [visibleContent, setVisibleContent] = useState(null)
 
-  const previousChildren = usePrevious(childrenState)
   const contentAnimation = useSpring(springContentAnimation)
   const modalAnimation = useSpring(springModalAnimation)
 
+  const handleEscapeKey = event => {
+    if (event && event.key === "Escape") {
+      onClose()
+    }
+  }
+
+  const updateEscapeKeyListener = () => {
+    if (show) {
+      document.addEventListener(KEYBOARD_EVENT, handleEscapeKey, true)
+    } else {
+      document.removeEventListener(KEYBOARD_EVENT, handleEscapeKey, true)
+    }
+  }
+
+  const updateBodyScrollBlock = () => {
+    if (show) {
+      document.body.style.overflowY = "hidden"
+    } else {
+      document.body.style.overflowY = "visible"
+    }
+  }
+
   useEffect(() => {
+    updateBodyScrollBlock()
+    updateEscapeKeyListener()
```

Comment:
It might be worth moving the contents of these functions inside of here with a comment, and consolidating around the `if (show)` conditional, rather than duplicating. It reads clear enough without the need for functions.